Update docs on how to rotate IAM keys (#2187)
alphagov/govuk-developer-docs: source/manual/aws-iam-key-rotation.html.md

for the environment you want to rotate keys for and you'll see a list of users
and the age of their access keys.

### 1. Find users access key

> Click on the user whose key you want to rotate, then click on “Security Credentials” to see their access key.

Reviewer: ```suggestion
Click on the user whose key you want to rotate.
Click on “Security Credentials” to see their access key.
Make a note of the Access Key ID.
```

### 2. Find where the key is used

Make a note of the Access Key ID. Enter `govuk-secrets` in your machine and `cd` into either `puppet` or `puppet_aws` find where this access key is defined in
[govuk-secrets](https://github.com/alphagov/govuk-secrets). You can find this Access Key ID by running one of the

> [Puppet AWS common actions](https://github.com/alphagov/govuk-secrets/tree/master/puppet_aws#common-actions) in both directories `puppet` and `puppet_aws` then searching for the key in the result. The command you should run is related to the environment you need to target. For example, to edit global staging credentials in the `govuk::apps` namespace, you need to run `bundle exec rake 'eyaml:edit[staging,apps]'`.

Reviewer: We can simplify this bit:
```
Enter [govuk-secrets/puppet](https://github.com/alphagov/govuk-secrets/puppet) in your machine.
Run one of the [Puppet AWS common actions](https://github.com/alphagov/govuk-secrets/tree/master/puppet_aws#common-actions) and search for the `Access Key ID` in the result.

> **Note**
> The command you should run is related to the environment you need to target.
> For example, to edit global staging credentials in the `govuk::apps` namespace, you need to run `bundle exec rake 'eyaml:edit[staging,apps]'`.

Now enter [govuk-secrets/puppet_aws](https://github.com/alphagov/govuk-secrets/puppet_aws), run the same command again and search for the Access Key ID in the result.
```

> Keys may also be defined in more then one place, for example the hieradata values

Reviewer: We can simplify this bit:
```
Enter [govuk-secrets/puppet](https://github.com/alphagov/govuk-secrets/puppet) in your machine.
Run one of the [Puppet AWS common actions](https://github.com/alphagov/govuk-secrets/tree/master/puppet_aws#common-actions) and search for the `Access Key ID` in the result.

> **Note**
> The command you should run is related to the environment you need to target.
> For example, to edit global staging credentials in the `govuk::apps` namespace, you need to run `bundle exec rake 'eyaml:edit[staging,apps]'`.

Now enter [govuk-secrets/puppet_aws](https://github.com/alphagov/govuk-secrets/puppet_aws), run the same command again and search for the Access Key ID in the result.
```

`govuk::apps::support::aws_access_key_id` and
`govuk::apps::support_api::aws_access_key_id` contain the same key.
Keys may also be shared across AWS and Carrenza environments.

If the key is not present in the hieradata, it might be present in the control panel of one of the 3rd party services, such as Fastly or Logit.

### 3. Create and deploy new key

> Now create a new access key by clicking the `Create access key` button in `Security credentials` of the Amazon IAM console.

Reviewer: ```suggestion
Click the `Create access key` button in `Security credentials` of the Amazon IAM console.
```

> Next step is to replace the old `Access Key ID` and `Secret access key`

Reviewer: ```suggestion
Replace the old `Access Key ID` and `Secret access key`
```

in [govuk-secrets](https://github.com/alphagov/govuk-secrets/tree/master) or the

> 3rd party service with the new `Access Key ID` and `Secret access key` you created.

Reviewer: ```suggestion
3rd party service with the new ones.
```

If the change is in `govuk-secrets`, then
[govuk-puppet](https://github.com/alphagov/govuk-puppet) will need to be redeployed
to pick up the new keys. This may take up to 30 minutes to take effect or you can ssh

> into the machine that uses that key and then run `govuk_puppet --verbose`.

Reviewer: We can simplify this bit:
```
If the change is in `govuk-secrets`, redeploy [govuk-puppet](https://github.com/alphagov/govuk-puppet) so it picks up the new keys.
This may take up to 30 minutes to take effect. To avoid waiting, you can run `govuk_puppet --verbose` in the machine that uses that key.
```

for the environment you want to rotate keys for and you'll see a list of users
and the age of their access keys.

### 1. Find users access key

Click on the user whose key you want to rotate. Click on `Security Credentials`. Make a note of the Access Key ID.

### 2. Find where the key is used

> Enter [govuk-secrets/puppet](https://github.com/alphagov/govuk-secrets/puppet) in your machine.

Reviewer 2: This link didn't work for me but https://github.com/alphagov/govuk-secrets/tree/master/puppet does

Author: Updated 👍

> Enter [govuk-secrets/puppet](https://github.com/alphagov/govuk-secrets/puppet) in your machine.

Reviewer: ```suggestion
Enter [govuk-secrets/puppet](https://github.com/alphagov/govuk-secrets/tree/master/puppet) in your machine.
```

Run one of the [Puppet AWS common actions](https://github.com/alphagov/govuk-secrets/tree/master/puppet_aws#common-actions) and search for the `Access Key ID` in the result.

> **Note**
> The command you should run is related to the environment you need to target.
> For example, to edit global staging credentials in the `govuk::apps` namespace, you need to run `bundle exec rake 'eyaml:edit[staging,apps]'`.

> Now enter [govuk-secrets/puppet_aws](https://github.com/alphagov/govuk-secrets/puppet_aws), run the same command again and search for the Access Key ID in the result.

Reviewer: ```suggestion
Now enter [govuk-secrets/puppet_aws](https://github.com/alphagov/govuk-secrets/tree/master/puppet_aws), run the same command again and search for the Access Key ID in the result.
```

### 2. Create and deploy new key

Now create a new access key by clicking the `Create access key` button in `Security credentials` of the IAM console. 
Next step is to update the Access Key ID and Secret access key
in [govuk-secrets](https://github.com/alphagov/govuk-secrets/tree/master) or the
3rd party service. 

> This is by running a command where the User is located from 

Reviewer 3: I'm a little unclear what you mean by this sentence
